fault/tester.py

```diff
@@ -536,6 +536,26 @@ def internal(self, *args):
         # return a SelectPath containing the desired path
         return SelectPath([self.circuit] + list(args))
 
+    def __call__(self, *args, **kwargs):
```

Comment:
Feel like the interface to this function can just be to return the references to the outputs, not to a `peek`. That avoids having to modify the `Peek` class interface. Adding a back-reference to a `Tester` object and a `expect()` function on a `Peek` instance seems a little out of place and clunky (and avoids making an breaking API change).

We could even return the "view" on those ports so you can still call `expect` on them:

```python
tester.circuit.O.expect(x)
tester(a, b).expect(y)
```

(A third option would be to **extend** the `Expect` class interface to take in a `Peek` object (instead of a port), but given that we have the port-view's as above, this seems suboptimal.)

Reply:
I'm not sure if it's worth the extra complexity?

I think doing something like
```
tester.peek(circuit.O).expect(3)
```
doesn't seem to unreasonable of thing to support.

In fact, you can already write
```
tester.circuit.O.expect(3)
```
but the wrapper machinery in the setattr interface already handles this, so it could be simplified if a peek object supported expect.

Reply:
My main point is that the actions should be generic objects that are independent of anything to do with the Tester class, which is why they're defined as such, and the Tester class consumes actions and does stuff with them.

Can't we get the same functionality out of call with returning the view object? So

```python = 
result = tuple(getattr(self.circuit, name) for name in self._circuit.interface.inputs_by_name())
```

and then we still have the same syntax in the calling code:

```python
    tester = fault.Tester(AndCircuit)
    for i, j in zip(range(2), range(2)):
        tester(i, j).expect(i & j)
```

since `tester(i, j)` returns a port-view object

Reply:
(Perhaps my first comment was ambiguous, by "view" object I mean the existing view objects. The fact that we can already do `tester.circuit.O.expect(3)` is exactly what I'm suggesting we leverage)